agent/lib/kontena/workers/service_pod_worker.rb

```diff
@@ -37,6 +39,10 @@ def ensure_desired_state
       exclusive {
         debug "state of #{service_pod.name}: #{service_pod.desired_state}"
         service_container = get_container(service_pod.service_id, service_pod.instance_number)
+        if service_container && !service_pod.running?
```

Comment:
The `HealthCheckWorker` could theoretically be running even if the `service_container` is nil? Unlikely, but this would be more robust if you leave off the `service_container` chckec...